### DISCUSSION TITLE
Bright ideas sought: detecting uasyncio

### DISCUSSION THREAD

**Discussion by peterhinch** [micropython/micropython#11560] [Ideas]
I raised this as an issue https://github.com/micropython/micropython/issues/11530 but I figured that not everyone sees these.

I'm after a function which can detect whether a piece of synchronous code was called from synchronous code or from a task. This is my best effort:
```py
def asyncio_running():
    try:
        z = asyncio.current_task()
    except:
        return False
    return True
```
It works, but as the issue reports, `uasyncio` retains state. If I run an application which uses `uasyncio`, then at the REPL, run code which uses that function, it will return `True` regardless of the facts.

Can anyone think of a non-hacky workround?

**Comment by jimmo** [comment]
I reckon we should fix this in uasyncio. See #11562
